src/gui/tray/UserModel.cpp

```diff
@@ -105,8 +105,65 @@ void User::slotBuildNotificationDisplay(const ActivityList &list)
 
 void User::setNotificationRefreshInterval(std::chrono::milliseconds interval)
 {
-    qCDebug(lcActivity) << "Starting Notification refresh timer with " << interval.count() / 1000 << " sec interval";
-    _notificationCheckTimer.start(interval.count());
+    if (!slotConnectToPushNotifications()) {
+        qCDebug(lcActivity) << "Starting Notification refresh timer with " << interval.count() / 1000 << " sec interval";
+        _notificationCheckTimer.start(interval.count());
+    }
+}
+
+bool User::slotConnectToPushNotifications()
+{
+    if (!pushNotificationsActivitiesReady()) {
+        // push notifications are not ready yet, so we'll subscribe and continue polling
+        connect(_account->account().data(), &Account::pushNotificationsReady, this, &User::slotConnectToPushNotifications, Qt::UniqueConnection);
+        return false;
+    }
+
+    if (qobject_cast<Account*>(sender()) == _account->account().data()) {
```

Comment:
This will be either true or `sender() == nullptr` (that is not used as a slot but as a regular method).

I'm wondering if those two aspects shouldn't be separated because everything but that if is about checking if the push notifications are ready or not, and then that if is about reacting to the push notifications becoming ready. So I'd put the content of  that if in its own slot and connect to that slot in the if above (since it's the only place we connect to it anyway).

I think this would improve readability quite a bit by separating concerns.